### PR TITLE
[examples] Fix errors detected by "openapi-diff" (aka "@azure/oad")

### PR DESCRIPTION
**Pull request by mikeharder**
Potential fixes we (or spec owners) could make to fix errors detected by tool `openapi-diff` (aka `@azure/oad`).

### CODE REVIEW

**Comment by timotheeguerin** [comment]
Not sure what is the type of the headers, string or number?

**Review by konrad-jamrozik — APPROVED** [review]

**Comment by mikeharder** [comment]
> Not sure what is the type of the headers, string or number?

I'm not sure exactly how to interpret schema, but maybe this means it must be string?

```json
    "header": {
      "type": "object",
      "additionalProperties": false,
      "required": ["type"],
      "properties": {
        "type": {
          "type": "string",
          "enum": ["string", "number", "integer", "boolean", "array"]
        },
        "format": {
          "type": "string"
        },
```

https://github.com/Azure/autorest/blob/da69517e79b8c2a1040c90313fbf9b10fdd1b46a/schema/swagger-extensions.json#L428-L439

Appears to be the same in autorest-core@v3:

https://github.com/Azure/autorest/blob/63ffe68961e24ed8aa59a2ca4c16a8019c271e45/packages/libs/autorest-schemas/swagger-extensions.json#L431-L442

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/examples/KustoPoolDataConnectionsUpdate.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
   "responses": {
     "200": {
       "headers": {
-        "location": "West US"
+        "location": "http://azure.async.operation/status"
```
Comment (mikeharder) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/examples/KustoPoolDataConnectionsUpdate.json:22]: Schema:

```json
  "xmsResponseHeaderExample": {
    "type": "object",
    "description": "Examples of response headers. Common response headers like 'location', 'azure-asyncoperation', 'retry-after', etc. are defined as optional properties for ease of use. Please feel free to add other response headers as applicable in the example.",
    "properties": {
      "location": {
        "type": "string",
        "description": "An example of the location header sent by the server, if applicable in the example.",
        "minLength": 10,
        "pattern": "^(http|https)://.*"
      },
```
https://github.com/Azure/autorest/blob/da69517e79b8c2a1040c90313fbf9b10fdd1b46a/schema/example-schema.json#L61-L70

**Review by timotheeguerin — APPROVED** [review]

**Comment by mikeharder** [comment]
@timotheeguerin: Do you think fixing the examples is the right approach here?  If so there may be more instances I will add to this PR.

Do you think we should get review from the owner of each example we are fixing?  Or are these changes minimal enough we just make them ourselves without review?

**Comment by timotheeguerin** [comment]
> @timotheeguerin: Do you think fixing the examples is the right approach here? If so there may be more instances I will add to this PR.
> 
> Do you think we should get review from the owner of each example we are fixing? Or are these changes minimal enough we just make them ourselves without review?

It looks to me like those examples are incorrect so I do feel like fixing them would be good. However yeah it may be better to get review from the owners. Do we need to fix all those issues though, or are those just going to start failing correctly next time they do an update of this service?

**Comment by mikeharder** [comment]
> It looks to me like those examples are incorrect so I do feel like fixing them would be good. However yeah it may be better to get review from the owners. Do we need to fix all those issues though, or are those just going to start failing correctly next time they do an update of this service?

These are failures I detected by running `openapi-diff` on every spec in main (diffing each spec against itself).  So you're correct we don't need to fix these now, we can leave these to the spec authors to fix in their next update.

A few more things I do want to check:

1. How did the incorrect examples get generated in the first place?  Were they hand-edited, or its this a problem with the tool we use to generate examples?

2. How did the PRs get merged to main if `autorest` was failing with invalid schema?  Might have happened during the window where `autorest` failures were being silently ignored.

**Review by konrad-jamrozik — APPROVED** [review]

**Pull request closed without merging**